### PR TITLE
added onClose option to handle disconnects

### PR DESCRIPTION
**Pull request by deyhle**
This pull request addresses #13 and adds an optional parameter to provide a disconnect handler.

### CODE REVIEW

File: lib/connection.js [hunk 1]
```diff
@@ -117,6 +118,14 @@ function Connection(options) {
     }
 
     /***
+     * Called when the socket is closed (does nothing)
+     * @param had_error true if the socket had a transmission error
+     */
+    function defaultOnClose(had_error) {
```
Comment (x-cubed) [inline lib/connection.js]: Fix up the naming style here: had_error => hadError

File: test/connection.js [hunk 2]
```diff
@@ -33,6 +33,21 @@ describe('Connection', function() {
                 assert.fail("Should not have been able to send a ping across a failed connection");
             });
         });
+
+        it('should call close event handler', function(done) {
+            var options = {
+                host: defaultHostName,
+                onError: done,
+                onClose: function (had_error) {
+                    assert.notEqual(had_error, true);
```
Comment (x-cubed) [inline test/connection.js]: Likewise: had_error => hadError

**Comment by x-cubed** [comment]
Looks good, just fix up the naming.
Thanks

**Comment by deyhle** [comment]
good idea, done!

**Pull request merged**